generate_report_ui.gs

```diff
@@ -0,0 +1,263 @@
+  if(e.formInput && e.formInput.projectFilter == "SPECIFIC_PROJECT") {
+    var projectId = CardService.newTextInput()
+    .setFieldName("projectId")
+    .setTitle("Enter the Cloud Project Id");
+    projectFiltersection.addWidget(projectId);
+  }
+  card.addSection(projectFiltersection);
+  
+  // Add the report type Filter
+  var reportTypesection = CardService.newCardSection().setHeader("<b>REPORT FILTER</b>").setCollapsible(true);
+  var reportType = CardService.newSelectionInput()
+    .setType(CardService.SelectionInputType.RADIO_BUTTON)
+    .setFieldName("reportType");
+  
+  // add items in the radio button
+   if(e.formInput) {
+    reportType.addItem("Most Executed Script", "mostExecutedScript", e.formInput.reportType == "mostExecutedScript")
+    reportType.addItem("Most Active Users", "mostActiveUsers", e.formInput.reportType == "mostActiveUsers")
+    reportType.addItem("Owners of Scripts", "ownerOfScripts", e.formInput.reportType == "ownerOfScripts");
+   }
+  else {
+    reportType.addItem("Most Executed Script", "mostExecutedScript", true)
+    reportType.addItem("Most Active Users", "mostActiveUsers", false)
+    reportType.addItem("Owners of Scripts", "ownerOfScripts", false);
+  }
+  reportType.setOnChangeAction(CardService.newAction()
+        .setFunctionName("reportTypeCallback"));
+  reportTypesection.addWidget(reportType);
+  
+  if(!e.formInput || e.formInput.reportType != "ownerOfScripts") {
+  var limitResponse = CardService.newTextInput()
+    .setFieldName("limitResponse")
+    .setTitle("Enter the Number Of Responses ");
+    reportTypesection.addWidget(limitResponse);
+  }
+  card.addSection(reportTypesection);
+  
+  var section2 = CardService.newCardSection();
+  // Create text Paragraph for placing url of report
+  if(reportUrlVal) {
+    var reportUrl = CardService.newTextParagraph()
+    .setText("<b>Report Url</b>\n <a href='"+reportUrlVal+"'>"+reportUrlVal+"</a>");
+    section2.addWidget(reportUrl);
+  }
+    
+  // Create a button for generating report
+  var getReportAction = CardService.newAction()
+      .setFunctionName('generateReport'); 
+  var getReport = CardService.newTextButton()
+      .setText('Generate Report')
+      .setOnClickAction(getReportAction)
+      .setTextButtonStyle(CardService.TextButtonStyle.FILLED);
+  section2.addWidget(getReport);
+  card.addSection(section2);
+  
+   return card.build();
+}
+
+/**
+* Callback function on clicking the GET REPORT button and create a card with most executed script and most active users
+* @return {CardService.Card} The card to show to the user
+*/
+function generateReport(e) {
+  var title, projTitle;
+  var toTime = new Date();
+  var fromTime;
+  var millisInBetween;
+  var input  = e.formInput;
+  var projectFilter = input.projectFilter;
+  var reportType = input.reportType;
+  var limitResponse = input.limitResponse;
+  if(!limitResponse) limitResponse = 1;
+    
+  if(input.timeFilter == "LAST_HOUR") { 
+    title = "Last Hour";
```

Comment:
rename title to timestamp_header or similar to it

Reply:
Updated